Add type file and missing files

berkbaski opened the pull request:
Hey!

I want to do two more fixes.

### 1. Updating the JavaScript samples in the README.md file to TypeScript.
In the examples on the README.md file, the types are given as characters and comment lines. I don't know how to update this.

Like this
```javascript
spriteObject.setSprite({
	sheet: Image.createFromFile("images://spritesheet.png"),
	frameX: << #frames in axis-X >>, // Number
	frameY: << #frames in axis-Y >>, // Number
	frameCount: << #frames in the sheet >> // Number
});
```

### 2. I want to make the parameter in the constructor optional if necessary. 
But I think incoming parameters pass to the Image object. I wasn't sure if I should do this.

I have time for suggestions.

berkbaski committed:
Added .npmrc

berkbaski committed:
Added .gitignore

berkbaski committed:
Added sf-core as peerDependency

berkbaski committed:
Added index.d.ts

furkanarabaci commented:
For 1#, I think it is better to leave the code snippet as is. Only change requires to import and it will be enough.

For 2#, if the parameters are passed to image object, then it is better to have them as optional.

Btw when I talk about constructor, I mean this:
`const spriteView = new SpriteView({// Parameters here});`

You should also have copy-paste ready snippet like this:

```
import SpriteView from 'sf-extension-spriteview';
cons spriteView = new SpriteView(); // Constructor parameters should be typed as optional, if there is any
spriteView.setSprite({
        sheet: Image.createFromFile("images://spritesheet.png"),
	frameX: 30, // This needs a brief explanation in documents
	frameY: 30, // This needs a brief explanation in documents
	frameCount: 24 // This needs a brief explanation in documents
});
```

The parameter in setSprite shouldn't be optional. I think every property in options parameter is required.

berkbaski committed:
Added constructor fields & constructor is optional

berkbaski committed:
Updated README.md

berkbaski committed:
Deleted whitespaces

berkbaski commented:
Okay, thank you for your suggestions.

1# I updated with your suggestion. By the way, I added the optional fields.
2# I tested your suggestion by IDE. Cool, didn't break! But not displaying if you didn't set the width, height fields. I should give a default value to these fields maybe?

berkbaski committed:
Updated constructor params

berkbaski committed:
Merge branch 'feature/typing-and-doc' of https://github.com/berkbaski/sf-extension-spriteview into feature/typing-and-doc

berkbaski commented on index.d.ts:
```diff
@@ -0,0 +1,114 @@
+     */
+    frameX: number,
+    /**
+     * Y Position of the window
```
Frame Y count
Distinct image count on Y axis

berkbaski commented on index.d.ts:
```diff
@@ -0,0 +1,114 @@
+     */
+    sheet: Image,
+    /**
+     * X Position of the window
```
Frame X count
Distinct image count on X axis

berkbaski commented on index.d.ts:
```diff
@@ -0,0 +1,114 @@
+    /**
+     * Frame count of the image
+     */
+    frameCount: number
```
Kill it!

berkbaski replied:
@furkanarabaci have an idea here? Sometimes I want to spoil the animation, maybe? I tested it and it shows as many frames as given.

furkanarabaci replied:
Make frameCount optional with default value to x*y

berkbaski replied:
Okay, did it!

berkbaski committed:
Updated fields descriptions

furkanarabaci reviewed (changes requested):
FIX THIS THING OR RETURN TO YOUR GODDAMN VILLAGE!

berkbaski committed:
Updated frameCount
